cpp/src/Glacier2/Glacier2Router.cpp

```diff
@@ -350,7 +350,7 @@ RouterService::start(int argc, char* argv[], int& status)
     // The session router is used directly as a servant for the main
     // Glacier2 router Ice object.
     //
-    Glacier2::RouterPrx routerPrx(clientAdapter->add(_sessionRouter, {"router", instanceName}));
+    Glacier2::RouterPrx routerPrx = clientAdapter->add<Glacier2::RouterPrx>(_sessionRouter, {"router", instanceName});
```

Comment:
Maybe we should use `auto` to not repeat the type name in the declaration and the template parameter.

Reply:
Yes. I believe I've fixed them all.